### ISSUE TITLE
Allmon3 Locked Up

### ISSUE THREAD

**Issue by tsawyer** [AllStarLink/Allmon3#172]
Allmon was locked up this morning. Here's the daemon log from a little before and a little after restarting. 

```
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18928' coro=<WebSocketServer._close() running at /usr/lib/python3/dist-packages/websockets/legacy/server.py:771>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18929' coro=<WebSocketServer._close() running at /usr/lib/python3/dist-packages/websockets/legacy/server.py:771>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18930' coro=<WebSocketServer._close() running at /usr/lib/python3/dist-packages/websockets/legacy/server.py:771>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18931' coro=<WebSocketServer._close() running at /usr/lib/python3/dist-packages/websockets/legacy/server.py:771>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task exception was never retrieved
Oct 15 07:23:25 Allmon allmon3[881599]: future: <Task finished name='Task-1' coro=<start_tasks() done, defined at /usr/bin/allmon3:79> exception=SystemExit(0)>
Oct 15 07:23:25 Allmon allmon3[881599]: Traceback (most recent call last):
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/bin/allmon3", line 247, in <module>
Oct 15 07:23:25 Allmon allmon3[881599]:     asyncio.run(start_tasks())
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/asyncio/runners.py", line 47, in run
Oct 15 07:23:25 Allmon allmon3[881599]:     _cancel_all_tasks(loop)
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/asyncio/runners.py", line 63, in _cancel_all_tasks
Oct 15 07:23:25 Allmon allmon3[881599]:     loop.run_until_complete(
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/asyncio/base_events.py", line 629, in run_until_complete
Oct 15 07:23:25 Allmon allmon3[881599]:     self.run_forever()
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/asyncio/base_events.py", line 596, in run_forever
Oct 15 07:23:25 Allmon allmon3[881599]:     self._run_once()
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/asyncio/base_events.py", line 1890, in _run_once
Oct 15 07:23:25 Allmon allmon3[881599]:     handle._run()
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/asyncio/events.py", line 80, in _run
Oct 15 07:23:25 Allmon allmon3[881599]:     self._context.run(self._callback, *self._args)
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/bin/allmon3", line 114, in start_tasks
Oct 15 07:23:25 Allmon allmon3[881599]:     await asyncio.gather(*tasks)
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
Oct 15 07:23:25 Allmon allmon3[881599]:     return loop.run_until_complete(main)
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/asyncio/base_events.py", line 629, in run_until_complete
Oct 15 07:23:25 Allmon allmon3[881599]:     self.run_forever()
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/asyncio/base_events.py", line 596, in run_forever
Oct 15 07:23:25 Allmon allmon3[881599]:     self._run_once()
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/asyncio/base_events.py", line 1890, in _run_once
Oct 15 07:23:25 Allmon allmon3[881599]:     handle._run()
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/asyncio/events.py", line 80, in _run
Oct 15 07:23:25 Allmon allmon3[881599]:     self._context.run(self._callback, *self._args)
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3/dist-packages/asl_allmon/node_db/__init__.py", line 82, in db_updater
Oct 15 07:23:25 Allmon allmon3[881599]:     self.get_allmon_db()
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3/dist-packages/asl_allmon/node_db/__init__.py", line 37, in get_allmon_db
Oct 15 07:23:25 Allmon allmon3[881599]:     dbf = response.read().decode("UTF-8")
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/http/client.py", line 465, in read
Oct 15 07:23:25 Allmon allmon3[881599]:     return self._readall_chunked()
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/http/client.py", line 575, in _readall_chunked
Oct 15 07:23:25 Allmon allmon3[881599]:     value.append(self._safe_read(chunk_left))
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/http/client.py", line 612, in _safe_read
Oct 15 07:23:25 Allmon allmon3[881599]:     data = self.fp.read(amt)
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/lib/python3.9/socket.py", line 704, in readinto
Oct 15 07:23:25 Allmon allmon3[881599]:     return self._sock.recv_into(b)
Oct 15 07:23:25 Allmon allmon3[881599]:   File "/usr/bin/allmon3", line 31, in sigterm_handler
Oct 15 07:23:25 Allmon allmon3[881599]:     sys.exit(0)
Oct 15 07:23:25 Allmon allmon3[881599]: SystemExit: 0
Oct 15 07:23:25 Allmon systemd[1]: allmon3.service: Succeeded.
Oct 15 07:23:25 Allmon systemd[1]: Stopped Allmon3 daemon.
Oct 15 07:23:25 Allmon systemd[1]: allmon3.service: Consumed 5h 26min 25.564s CPU time.
Oct 15 07:23:25 Allmon systemd[1]: Starting Allmon3 daemon...
Oct 15 07:23:25 Allmon systemd[1]: Started Allmon3 daemon.
Oct 15 07:23:25 Allmon allmon3[936613]: INFO: (asl_allmon.web_configs) Allmon3 Master HTTP port is 16080
Oct 15 07:23:25 Allmon allmon3[936613]: INFO: (asl_allmon.web_configs) Allmon3 websockets starting port is 16700
Oct 15 07:23:25 Allmon allmon3[936613]: INFO: (asl_allmon.web_configs) Binding sevices to all addresses
Oct 15 07:23:25 Allmon allmon3[936613]: INFO: (asl_allmon.node_db) Retrieving database from http://allmondb.allstarlink.org/
Oct 15 07:23:26 Allmon allmon3[936613]: INFO: (asl_allmon.node_db) Updated node database in 0.68 seconds
Oct 15 07:23:26 Allmon allmon3[936613]: INFO: (allmon3) starting status ws 2509@2509.nodes.allstarlink.org on port 16700
```

**Comment by jxmx** [comment]
@tsawyer  is there more further back? That's almost looks like something tried to kill the allmon3 process before that.

**Comment by tsawyer** [comment]
There are a whole bunch of these...
```
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18932' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: /usr/lib/python3.9/asyncio/base_events.py:667: RuntimeWarning: coroutine 'BaseSelectorEventLoop._accept_connection2' was never awaited
Oct 15 07:23:25 Allmon allmon3[881599]:   self._ready.clear()
Oct 15 07:23:25 Allmon allmon3[881599]: RuntimeWarning: Enable tracemalloc to get the object allocation traceback
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18933' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18934' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18935' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18936' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18937' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18938' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18939' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18940' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18941' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18942' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18943' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18944' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18945' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18946' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18947' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18948' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
Oct 15 07:23:25 Allmon allmon3[881599]: task: <Task pending name='Task-18949' coro=<BaseSelectorEventLoop._accept_connection2() running at /usr/lib/python3.9/asyncio/selector_events.py:198>>
Oct 15 07:23:25 Allmon allmon3[881599]: Task was destroyed but it is pending!
```

**Comment by jxmx** [comment]
This still doesn't show the root of the problem.

**Comment by jxmx** [comment]
There's nothing I can do with this report.

**Comment by tsawyer** [comment]
See if this helps.
[allmon3_awp.log.zip](https://github.com/AllStarLink/Allmon3/files/13210525/allmon3_awp.log.zip)

**Comment by jxmx** [comment]
@tsawyer  Yes that was helpful and identical to the bug I just fixed in #171
